### ISSUE TITLE
Deconstruct() not invoked when right-hand-side is a tuple type

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#16190]
According to [the specification](https://github.com/dotnet/roslyn/blob/master/docs/features/deconstruction.md)

> In short, what this does in the general case is find a `Deconstruct` method on the expression on the right-hand-side of the assignment, invoke it with the appropriate number of `out var` parameters, converts those output values (if needed) and assign them to the variables on the left-hand-side. And in the special case where the expression on the right-hand-side is a tuple (tuple expression or tuple type), then the elements of the tuple can be assigned to the variables on the left-hand-side without needing to call `Deconstruct`.

The following test should pass (although the special case applies, so that it *can* be done without invoking `Deconstruct`, in this case the types match only when calling `Deconstruct`). But it fails.

To be clear, this may be a problem with the specification or it may be a problem with the implementation.

```cs
        [Fact]
        public void CallDeconstructForTuple()
        {
            string source = @"
class Program
{
    public static void Main()
    {
        (int x, int y, int z) = (1, 2);
    }
}
static class Extensions
{
    public static void Deconstruct(this (int, int) self, out int x, out int y, out int z)
    {
        x = self.Item1;
        y = self.Item2;
        z = 0;
    }
}
";
            var comp = CreateCompilationWithMscorlibAndSystemCore(source,
                references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
                options: TestOptions.DebugExe);
            comp.VerifyDiagnostics(
                );
        }
```

**Comment by jcouv** [comment]
This is a spec problem, maybe I didn't phrase it well. 

How about:
And in the special case where the expression on the right-hand-side is a tuple (tuple expression or tuple type), then the elements of the tuple ~~can be~~ *are* assigned to the variables on the left-hand-side without needing to call Deconstruct.

**Comment by gafter** [comment]
@jcouv I think a bit more is needed. For example, should that `And` be a `But`?

**Comment by jcouv** [comment]
Fixed doc with #16643
